python/lsst/daf/butler/_limited_butler.py

```diff
@@ -273,6 +276,52 @@ def getDeferred(
         """
         return DeferredDatasetHandle(butler=self, ref=ref, parameters=parameters, storageClass=storageClass)
 
+    def stored(self, ref: DatasetRef) -> bool:
+        """Indicate whether the dataset is present in the Datastore.
+
+        Parameters
+        ----------
+        ref : `DatasetRef`
+            Resolved reference to a dataset.
+
+        Returns
+        -------
+        stored : `bool`
+            Whether the dataset exists in the datastore and can be retrieved.
+        """
+        return self.datastore.exists(ref)
+
+    def stored_many(
+        self,
+        refs: Iterable[DatasetRef],
+        artifact_existence: Optional[dict[ResourcePath, bool]] = None,
```

Comment:
I am torn as to whether `artifact_existence` should be a parameter here. It's useful as an opaque cache if people know they are going to call it many times but only if they also know that nothing is actively writing files between calls to it (which `transfer_from` does assume). I think I'm talking myself out of having it here as being too dangerous.

Reply:
I think I've talked myself out of forwarding `artifact_existence` for the above reasons.